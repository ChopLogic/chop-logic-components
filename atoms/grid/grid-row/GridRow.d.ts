import { default as React } from '../../../../node_modules/react';
type Props = {
    rowId: string;
    selectedIds: string[];
    selectRowById: (id: string) => void;
    deselectRowById: (id: string) => void;
    selectable: boolean;
    values: (string | React.ReactElement)[];
    disabled?: boolean;
};
export declare const GridRow: React.FC<Props>;
export {};
