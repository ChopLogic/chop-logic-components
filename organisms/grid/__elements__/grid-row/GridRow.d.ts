import { FC, ReactElement } from '../../../../../node_modules/react';
type Props = {
    rowId: string;
    selectedIds: string[];
    selectRowById: (id: string) => void;
    deselectRowById: (id: string) => void;
    selectable: boolean;
    values: (string | ReactElement)[];
    disabled?: boolean;
};
export declare const GridRow: FC<Props>;
export {};
