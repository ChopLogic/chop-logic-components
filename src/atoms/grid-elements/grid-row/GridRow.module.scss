.row {
  border-bottom: var(--cl-light-border);

  &__selected {
    background-color: var(--cl-highlight-background);
  }
}
