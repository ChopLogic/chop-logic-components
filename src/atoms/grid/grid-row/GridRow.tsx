import React from 'react';
import { GridDataCell } from '@atoms/grid/data-cell/GridDataCell.tsx';
import { SelectGridRowCell } from '@atoms/grid/select-grid-row-cell/SelectGridRowCell.tsx';
import { getClassName } from '@utils';
import styles from './GridRow.module.scss';

type Props = {
  rowId: string;
  selectedIds: string[];
  selectRowById: (id: string) => void;
  deselectRowById: (id: string) => void;
  selectable: boolean;
  values: (string | React.ReactElement)[];
  disabled?: boolean;
};

export const GridRow: React.FC<Props> = ({ rowId, selectRowById, deselectRowById, selectable, values, selectedIds, disabled }) => {
  const selected = selectedIds.includes(rowId);

  return (
    <tr className={getClassName([styles.row, { [styles.row__selected]: selected }])}>
      {selectable && (
        <SelectGridRowCell
          rowId={rowId}
          selectRowById={selectRowById}
          deselectRowById={deselectRowById}
          isRowSelected={selected}
          disabled={disabled}
        />
      )}
      {values.map((value, index) => (
        <GridDataCell key={`row_${index}`} value={value} />
      ))}
    </tr>
  );
};
