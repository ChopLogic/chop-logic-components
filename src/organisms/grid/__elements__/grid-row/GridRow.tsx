import { getClassName } from '@utils';
import React from 'react';

import { GridDataCell } from '../data-cell/GridDataCell';
import { SelectGridRowCell } from '../select-grid-row-cell/SelectGridRowCell';
import styles from './GridRow.module.scss';

type Props = {
  rowId: string;
  selectedIds: string[];
  selectRowById: (id: string) => void;
  deselectRowById: (id: string) => void;
  selectable: boolean;
  values: (string | React.ReactElement)[];
  disabled?: boolean;
};

export const GridRow: React.FC<Props> = ({ rowId, selectRowById, deselectRowById, selectable, values, selectedIds, disabled }) => {
  const selected = selectedIds.includes(rowId);

  return (
    <tr className={getClassName([styles.row, { [styles.row__selected]: selected }])}>
      {selectable && (
        <SelectGridRowCell
          rowId={rowId}
          selectRowById={selectRowById}
          deselectRowById={deselectRowById}
          isRowSelected={selected}
          disabled={disabled}
        />
      )}
      {values.map((value, index) => (
        <GridDataCell key={`row_${index}`} value={value} />
      ))}
    </tr>
  );
};
