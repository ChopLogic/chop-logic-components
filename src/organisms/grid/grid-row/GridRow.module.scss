.row {
  border-bottom: var(--cl-light-border);
  background-color: var(--cl-primary-background);

  &__selected {
    background-color: var(--cl-highlight-background);
  }
}
