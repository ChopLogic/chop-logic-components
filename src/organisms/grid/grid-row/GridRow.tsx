import { getClassName } from '@utils';
import { FC, ReactElement } from 'react';

import { GridDataCell } from '../data-cell/GridDataCell.tsx';
import { SelectGridRowCell } from '../select-grid-row-cell/SelectGridRowCell.tsx';
import styles from './GridRow.module.scss';

type Props = {
  rowId: string;
  selectedIds: string[];
  selectRowById: (id: string) => void;
  deselectRowById: (id: string) => void;
  selectable: boolean;
  values: (string | ReactElement)[];
  disabled?: boolean;
};

export const GridRow: FC<Props> = ({ rowId, selectRowById, deselectRowById, selectable, values, selectedIds, disabled }) => {
  const selected = selectedIds.includes(rowId);

  return (
    <tr className={getClassName([styles.row, { [styles.row__selected]: selected }])}>
      {selectable && (
        <SelectGridRowCell
          rowId={rowId}
          selectRowById={selectRowById}
          deselectRowById={deselectRowById}
          isRowSelected={selected}
          disabled={disabled}
        />
      )}
      {values.map((value, index) => (
        <GridDataCell key={`row_${index}`} value={value} />
      ))}
    </tr>
  );
};
