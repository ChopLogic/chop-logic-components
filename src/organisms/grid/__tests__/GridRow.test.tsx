import { GridRow } from '@organisms/grid/grid-row/GridRow';
import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

describe('GridRow', () => {
  const testProps = {
    rowId: 'item-1',
    selectRowById: vi.fn(),
    deselectRowById: vi.fn(),
    selectable: true,
    values: [
      { value: 'Value 1', field: 'col1' },
      { value: 'Value 2', field: 'col2' },
      { value: <span key='value-3'>Value 3</span>, field: 'col3' },
    ],
    selectedIds: ['value1', 'value2', 'value3'],
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <GridRow {...testProps} />
        </tbody>
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render a checkbox if the row is not selectable', () => {
    render(
      <table>
        <tbody>
          <GridRow {...testProps} selectable={false} />
        </tbody>
      </table>,
    );
    expect(screen.queryByRole('checkbox')).not.toBeInTheDocument();
  });
});
