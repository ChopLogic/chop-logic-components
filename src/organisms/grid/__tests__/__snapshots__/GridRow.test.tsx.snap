// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GridRow > should match the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="_row_2c96fd"
      >
        <td
          class="_cell_4fae7d"
        >
          <div
            class="_checkbox_a434a4"
          >
            <input
              class="_checkbox_input_a434a4"
              id="select_row_item-1"
              type="checkbox"
            />
            <label
              class="_label_c20562"
              for="select_row_item-1"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="_cell_ae74a7"
        >
          Value 1
        </td>
        <td
          class="_cell_ae74a7"
        >
          Value 2
        </td>
        <td
          class="_cell_ae74a7"
        >
          <span>
            Value 3
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
